### ISSUE TITLE
When creating a tag for shortcode, keep the underscore.

### ISSUE THREAD

**Issue by jimmyandrade** [fusioneng/shortcake-bakery#12]
I am creating a shortcode in which the class is `Scroll_Point`. The shortcode tag being generated is `scroll-point`, but it would be better if it was `scroll_point`.

Why? Because there are [recommendations about avoiding use of shortcode tags with the hyphen character](https://codex.wordpress.org/Shortcode_API#Hyphens) on WordPress Shortcode API docs.

So, I think it would be better not to replace the classes' underscores by hyphens, but keep 'em.

**Comment by danielbachhuber** [comment]
Hm, I'll have to give this some thought. I'm personally not a huge fan of underscores. It seems like the core bug should be fixed too.

**Comment by jimmyandrade** [comment]
Hm... on second thought, I think you're right. It's the responsibility of the WordPress core developers solve this hyphens issue. And particularly, I also prefer the hyphens. I look more aesthetically pleasing.

**Comment by danielbachhuber** [comment]
I'll keep this open until I track down the current state of underscores vs dashes in core.

**Comment by davisshaver** [comment]
@danielbachhuber Did you ever find current status in core? From [this ticket](https://core.trac.wordpress.org/ticket/17657), it appears that the issue may have been fixed. Maybe we just need to suggest a Codex update? cc @jimmyandrade

**Comment by danielbachhuber** [comment]
> From this ticket, it appears that the issue may have been fixed

Yep, I'd agree with that assessment. I haven't had the chance to look in further.

**Comment by jimmyandrade** [comment]
@davisshaver yep, I agree with this, too. cc @danielbachhuber
